### PR TITLE
[KB] SAML - Add auth method mismatch + consistency, procedural, language improvements

### PR DESCRIPTION
**Pull request by armchairlinguist**
We had some consistency issues with the SAML/SSO KBs such as different title formats, present or missing help, etc. Some also had a level of technical detail that didn't match the intended audience. This PR is an attempt to align them in various ways; I also added a new KB with another error that can come up and be addressed by adjusting config.

I opted not to change any filenames, to avoid changing the URLs, even though the titles are different.

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+![SAML auth method error](/img/kb/saml-auth-method-doesnt-match.png)
+
+This error only occurs when starting sign-in from Semgrep. To prevent the error, start your sign-in from your IdP (SSO provider).
+
+To fix this problem, you must be an `admin` in Semgrep AppSec Platform.
+
+1. Log in to [Semgrep AppSec Platform](https://semgrep.dev/login).
+2. Click **<i class="fa-solid fa-gear"></i> Settings > Access > [SSO](https://semgrep.dev/orgs/-/settings/access/sso)**.
+4. Check the box at the bottom labeled **Check if your SSO supports non-password authentication mechanisms (e.g. MFA, X509, PasswordLessPhoneSignin)**.
```
Comment (khorne3) [inline docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md:23]:
```suggestion
4. Check the box labeled **Check if your SSO supports non-password authentication mechanisms (e.g. MFA, X509, PasswordLessPhoneSignin)**.
```

File: docs/kb/semgrep-appsec-platform/saml-bad-signature.md [hunk 2]
```diff
@@ -28,4 +30,13 @@ Check information such as:
 - Validity dates
 - Signature algorithm and value
 
-Address any problems with the certificate. Then, ensure that you have uploaded the resulting certificate correctly to Semgrep AppSec Platform, and attempt a new SSO login.
+Address any problems with the certificate. Then, upload the resulting certificate to the Semgrep AppSec Platform:
```
Comment (khorne3) [inline docs/kb/semgrep-appsec-platform/saml-bad-signature.md:33]:
```suggestion
Address any problems with the certificate. Then, upload the resulting certificate to Semgrep AppSec Platform: 
```

File: docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+When logging in to Semgrep using SAML single-sign on (SSO), you may see an error `Authentication method doesn't match requested`. The error might look something like this:
+
+![SAML auth method error](/img/kb/saml-auth-method-doesnt-match.png)
+
+This error only occurs when starting sign-in from Semgrep. To prevent the error, start your sign-in from your IdP (SSO provider).
+
+To fix this problem, you must be an `admin` in Semgrep AppSec Platform.
+
+1. Log in to [Semgrep AppSec Platform](https://semgrep.dev/login).
```
Comment (khorne3) [inline docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md:21]:
```suggestion
1. Sign in to [Semgrep AppSec Platform](https://semgrep.dev/login).
```

Small thing, but we've opted for "Sign in" to reflect the UI language (though you might see strays as I change all of these over)

File: docs/kb/semgrep-appsec-platform/saml-bad-signature.md [hunk 4]
```diff
@@ -28,4 +30,13 @@ Check information such as:
 - Validity dates
 - Signature algorithm and value
 
-Address any problems with the certificate. Then, ensure that you have uploaded the resulting certificate correctly to Semgrep AppSec Platform, and attempt a new SSO login.
+Address any problems with the certificate. Then, upload the resulting certificate to the Semgrep AppSec Platform: 
+
+1. Log in to [Semgrep AppSec Platform](https://semgrep.dev/login).
```
Comment (khorne3) [inline docs/kb/semgrep-appsec-platform/saml-bad-signature.md:35]:
```suggestion
1. Sign in to [Semgrep AppSec Platform](https://semgrep.dev/login).
```

File: docs/kb/semgrep-appsec-platform/saml-bad-signature.md [hunk 5]
```diff
@@ -28,4 +30,13 @@ Check information such as:
...
+2. Click **<i class="fa-solid fa-gear"></i> Settings > Access > [SSO](https://semgrep.dev/orgs/-/settings/access/sso)**.
+3. In the **Upload/Paste certificate** box, add the correct certificate.
+4. Click **Save** to save this setting.
```
Comment (khorne3) [inline docs/kb/semgrep-appsec-platform/saml-bad-signature.md:38]:
```suggestion
4. Click **Save**.
```

File: docs/kb/semgrep-appsec-platform/saml-stops-working.md [hunk 6]
```diff
@@ -8,92 +8,31 @@ tags:
...
 
-## The SAML configuration uses a faulty SSL certificate
+## Certificate issues
 
-One component of the SAML configuration is host validation of the targeted IdP
-server, which entails uploading and registering the IdP's x509 certificate. If
-the certificate becomes invalid - expiration, revocation, or rotation - host
-validation fails for the IdP server. Then, the IdP server fails to respond to
-requests.
+If you see the error `Signature validation failed. SAML Response rejected` it's likely there is a problem with the certificate. If the IdP certificate becomes invalid via expiration, revocation, or rotation, signature validation fails.
```
Comment (khorne3) [inline docs/kb/semgrep-appsec-platform/saml-stops-working.md:17]:
```suggestion
If you see the error `Signature validation failed. SAML Response rejected` it's likely there is a problem with the certificate. The IdP certificate may be invalid due to expiration, revocation, or rotation, or signature validation fails.
```

**Review by khorne3 — APPROVED** [review]
Just some small things, but otherwise all good. Thank you!

File: docs/kb/semgrep-appsec-platform/saml-stops-working.md (hunk 6 above)
Comment (armchairlinguist) [reply docs/kb/semgrep-appsec-platform/saml-stops-working.md:17]: This is not semantically the same as what I said - the signature validation failure is a result of the other issues, not a cause of an issue. Skipping this one, feel free to adjust the wording later as long as the sense is preserved.

Comment (armchairlinguist) [reply docs/kb/semgrep-appsec-platform/saml-stops-working.md:17]: That's probably fine for this context. It fails if the cert becomes invalid for any reason, but those are the most common reasons. I'll make the change.

File: docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  - Semgrep AppSec Platform
+  - SSO
+  - SAML
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+# SAML SSO Error: Authentication method doesn't match requested
+
+When logging in to Semgrep using SAML single-sign on (SSO), you may see an error `Authentication method doesn't match requested`. The error might look something like this:
```
Comment (sinat101) [inline docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md:13]:
```suggestion
When logging in to Semgrep using SAML single-sign on (SSO), you may see the error `Authentication method doesn't match requested`:
```

**Review by sinat101 — APPROVED** [review]

File: docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md (hunk 7 above)
Comment (sinat101) [reply docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md:13]: I'm pretty ambivalent, we can keep or remove. I'd just change the "an" to a "the"

Comment (armchairlinguist) [reply docs/kb/semgrep-appsec-platform/saml-authentication-method-match.md:13]: yeah I liked that part of the change. I think I'm just going to commit your version for now

**Pull request merged**